http-scala-fx/src/main/scala/fx/Http.scala

```diff
@@ -0,0 +1,329 @@
+
+  @tailrec
+  private def GET[A](
+      retryCount: Int,
+      maybeTimeout: Option[ScalaDuration],
+      headers: HttpHeader*): HttpResponseMapper[A] ?=> Http[HttpResponse[A]] =
+    val mapper = summon[HttpResponseMapper[A]]
+    val config = summon[HttpClientConfig]
+    val retries = config.maximumRetries.getOrElse(summon[HttpRetries])
+    val requestBuilder =
+      addHeadersToRequestBuilder(HttpRequest.newBuilder().GET().uri(uri), headers: _*)
+
+    val response: HttpResponse[A] =
+      handle(
+        summon[HttpClient]
+          .client
+          .send(
+            maybeTimeout
+              .fold(requestBuilder) { fd =>
+                requestBuilder.timeout(Duration.ofMillis(fd.toMillis))
+              }
+              .build(),
+            mapper.bodyHandler)) { ex => HttpExecutionException(ex).shift }
+
+    if (response.shouldRetry(retryCount)) {
+      GET(retryCount + 1, maybeTimeout, headers: _*)
+    } else {
+      response
+    }
+
+  def POST[A, B](body: B, headers: HttpHeader*)
+      : (HttpResponseMapper[A], HttpBodyMapper[B]) ?=> Http[HttpResponse[A]] =
+    POST(body, None, 0, headers: _*)
+
+  def POST[A, B](body: B, timeout: ScalaDuration, headers: HttpHeader*)
+      : (HttpResponseMapper[A], HttpBodyMapper[B]) ?=> Http[HttpResponse[A]] =
+    POST(body, Option(timeout), 0, headers: _*)
+
+  @tailrec
+  private def POST[A, B](
+      body: B,
+      maybeTimeout: Option[ScalaDuration],
+      retryCount: Int = 0,
+      headers: HttpHeader*)
+      : (HttpResponseMapper[A], HttpBodyMapper[B]) ?=> Http[HttpResponse[A]] =
+    val mapper = summon[HttpResponseMapper[A]]
+    val bodyMapper = summon[HttpBodyMapper[B]]
+    val config = summon[HttpClientConfig]
+    val retries = config.maximumRetries.getOrElse(summon[HttpRetries])
+    val requestBuilder = addHeadersToRequestBuilder(
+      HttpRequest.newBuilder().POST(bodyMapper.bodyPublisher(body)).uri(uri),
+      headers: _*)
+
+    val response =
+      handle(
+        summon[HttpClient]
+          .client
+          .send(
+            maybeTimeout
+              .fold(requestBuilder) { fd =>
+                requestBuilder.timeout(Duration.ofMillis(fd.toMillis))
+              }
+              .build(),
+            mapper.bodyHandler)) { ex => HttpExecutionException(ex).shift }
+    if (response.shouldRetry(retryCount)) {
+      POST(body, maybeTimeout, retryCount + 1, headers: _*)
+    } else {
+      response
+    }
+
+  def HEAD(headers: HttpHeader*): Http[HttpResponse[Void]] = HEAD(None, 0, headers: _*)
+  def HEAD(timeout: ScalaDuration, headers: HttpHeader*): Http[HttpResponse[Void]] =
+    HEAD(Option(timeout), 0, headers: _*)
+
+  @tailrec
+  private def HEAD(
+      maybeTimeout: Option[ScalaDuration],
+      retryCount: Int = 0,
+      headers: HttpHeader*): Http[HttpResponse[Void]] =
+    val config = summon[HttpClientConfig]
+    val retries = config.maximumRetries.getOrElse(summon[HttpRetries])
```

Comment:
```suggestion
```

Reply:
Same as above.

Reply:
Resolved in [ce2b2a4](https://github.com/47deg/TBD/pull/37/commits/ce2b2a4da2a01b983a46e3a5ab23f80770d74d24)